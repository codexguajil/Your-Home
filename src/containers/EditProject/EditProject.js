import React from 'react';
import { Button, Modal, Form, Input } from 'antd';

const CollectionCreateForm = Form.create({ name: 'form_in_modal' })(
  class extends React.Component {
    render() {
      const { visible, onCancel, onCreate, form, editProject } = this.props;
      const { getFieldDecorator } = form;
      return (
        <Modal
          visible={visible}
          title="Edit Project Details"
          okText="Update"
          onCancel={onCancel}
          onOk={onCreate}
        >
          <Form layout="vertical">
            <Form.Item label="Name">
              {getFieldDecorator('name', {
                rules: [{ required: false }],
              })(<Input onChange={editProject} name='name' placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Address">
              {getFieldDecorator('address', {
                rules: [{ required: false }],
              })(<Input onChange={editProject} name='address' placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Description">
              {getFieldDecorator('description')(<Input onChange={editProject} name='description' type="textarea" placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item className="collection-create-form_last-form-item">
            </Form.Item>
          </Form>
        </Modal>
      );
    }
  }
);

class EditProject extends React.Component {
  state = { 
    visible: false,
    name: '',
    address: '',
    description: ''
  };

  showModal = () => {
    this.setState({ visible: true });
  }

  editProject = (e) => {
    this.setState({[e.target.name]: e.target.value});
  }

  handleCancel = () => {
    this.setState({ visible: false });
  }

  handleCreate = () => {
    const { name, address, description } = this.state;
    const updatedProject = { name, address, description };
    console.log(updatedProject);
    this.setState({ visible: false});
  }

  render() {
    return (
      <div>
        <Button type="primary" onClick={this.showModal}>
          <i className="fas fa-pen"></i>
        </Button>
        <CollectionCreateForm
          visible={this.state.visible}
          onCancel={this.handleCancel}
          onCreate={this.handleCreate}
          editProject={this.editProject}
        />
      </div>
    );
  }
}

export default EditProject;