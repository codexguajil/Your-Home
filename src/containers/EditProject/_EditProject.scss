.sub-title {
  margin-bottom: 0;
}

.project-edit {
  margin-left: 25px;
}