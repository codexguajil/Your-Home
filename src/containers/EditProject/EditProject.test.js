import React from 'react';
import { shallow } from 'enzyme';
import EditProject from './EditProject';

describe('EditProject', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EditProject/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('showModal', () => {
    wrapper.instance().showModal();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('handleCreate', () => {
    wrapper.instance().handleCancel();
    expect(wrapper.state('visible')).toBe(false);
  });

  it('handelCreate', () => {

  });

  it('saveFormRef', () => {

  });

});