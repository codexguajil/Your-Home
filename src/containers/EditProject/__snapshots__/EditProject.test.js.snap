// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProject should match snapshot 1`] = `
<div>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    <i
      className="fas fa-pen"
    />
  </Button>
  <Form(Component)
    editProject={[Function]}
    onCancel={[Function]}
    onCreate={[Function]}
    visible={false}
  />
</div>
`;
